db-engine/src/main/java/com/splicemachine/db/iapi/util/DecimalUtil.java

```diff
@@ -0,0 +1,130 @@
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.db.iapi.util;
+
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.reference.SQLState;
+import com.splicemachine.db.iapi.types.DataValueDescriptor;
+import com.splicemachine.db.iapi.types.DataValueFactoryImpl;
+import com.splicemachine.db.iapi.types.SQLDecfloat;
+import com.splicemachine.db.iapi.types.SQLDecimal;
+import com.splicemachine.primitives.Bytes;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public interface DecimalUtil {
+
+    /**
+     * Implements DB2 decimal function, for more information go to:
+     * https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000791.html
+     */
+    static DataValueDescriptor getDecimalDb2(DataValueDescriptor dvd,
+                                             int precision,
+                                             int scale,
+                                             String decimalCharacter) throws StandardException {
+        if (dvd == null || dvd.isNull()) {
+            return null;
+        }
+        BigDecimal bigDecimal = null;
+        Calendar cal = null;
+        switch (DataValueFactoryImpl.Format.formatFor(dvd)) {
+            case TINYINT: // fallthrough
+            case SMALLINT: // fallthrough
+            case INTEGER: // fallthrough
+            case LONGINT: {
+                if (String.valueOf(Math.abs(dvd.getLong())).length() > (precision - scale)) {
+                    throw StandardException.newException(SQLState.LANG_INVALID_DECIMAL_CONVERSION, dvd.toString(), precision, scale); // DB2 SQLSTATE 22003
+                }
+                bigDecimal = new BigDecimal(new BigInteger(Bytes.toBytes(dvd.getLong())), 0);
+            }
+            break;
+            case DECIMAL: {
+                bigDecimal = SQLDecimal.getBigDecimal(dvd);
+                if (bigDecimal == null) return null;
+            }
+            break;
+            case DECFLOAT: {
+                bigDecimal = SQLDecfloat.getBigDecimal(dvd);
+                if (bigDecimal == null) return null;
+            }
+            break;
+            case CHAR: // fallthrough
```

Comment:
Why not construct a SQLDecimal directly from the string?

Reply:
The string could contain invalid characters, it could also contain a comma for example and decimal separator(s), that's why it is not possible.